valhalla/sif/dynamiccost.h

```diff
@@ -569,58 +569,58 @@ class DynamicCost {
                                    const uint64_t current_time,
                                    const uint32_t tz_index,
                                    uint8_t& restriction_idx) const {
-    if (ignore_restrictions_ || !(edge->access_restriction() & access_mode))
+    if (!(edge->access_restriction() & access_mode))
       return true;
 
     const std::vector<baldr::AccessRestriction>& restrictions =
         tile->GetAccessRestrictions(edgeid.id(), access_mode);
 
-    bool time_allowed = false;
+    // do 2 loops to not disregard "maxaxles" (restrictions are sorted, see graphconstants.h)
+    for (const auto& restriction : restrictions) {
+      if (!ModeSpecificAllowed(restriction)) {
+        return false;
+      }
+    }
```

Comment:
this is unfortunately necessary, as `maxaxles` was added after the time restrictions and would be disregarded if a timed restriction triggers and returns early